README.md

```diff
@@ -10,28 +10,186 @@ The features are available before an official release so that users and collabor
+executed on device and in a vectorized fashion<sup>(2)</sup>, 
+which process and prepare the data coming out of the environments to be used by the agent:
+    <details>
+      <summary>Code</summary>
+    
+    ```python
+    env_make = lambda: GymEnv("Pendulum-v1", from_pixels=True)
+    env_base = ParallelEnv(4, env_make, device="cuda:0")  # creates 4 envs in parallel
+    env = TransformedEnv(
+        env_base, 
+        Compose(ToTensorImage(), ObservationNorm(loc=0.5, scale=1.0)),  # executes the transforms once and on device
+    )
+    tensordict = env.reset()
+    assert tensordict.device == torch.device("cuda:0")
+    ```
+    </details>
+
+- various tools for distributed learning (e.g. [memory mapped tensors](torchrl/data/tensordict/memmap.py))<sup>(2)</sup>;
+- various [architectures](torchrl/modules/models/) and models (e.g. [actor-critic](torchrl/modules/tensordict_module/actors.py))<sup>(1)</sup>:
+    <details>
+      <summary>Code</summary>
+    
+    ```python
+    common_module = ConvNet(
+        bias_last_layer=True,
+        depth=None,
+        num_cells=[32, 64, 64],
+        kernel_sizes=[8, 4, 3],
+        strides=[4, 2, 1],
+    )
+    common_module = TensorDictModule(
+        common_module,
+        in_keys=["pixels"],
+        out_keys=["hidden"],
+    )
+    policy_module = NormalParamsWrapper(
+        MLP(
+            num_cells=[64, 64],
+            out_features=32,
+            activation=nn.ELU,
+        )
+    )
+    policy_module = ProbabilisticTensorDict(  # stochastic policy
+        TensorDictModule(
+            policy_module,
+            in_keys=["hidden"],
+            out_keys=["loc", "scale"],
+        ),
+        dist_param_keys=["loc", "scale"],
+        out_key_sample="action",
+        distribution_class=TanhNormal,
+    )
+    value_module = MLP(
+        num_cells=[64, 64],
+        out_features=1,
+        activation=nn.ELU,
+    )
+    actor_value = ActorValueOperator(common_module, policy_module, value_module)
+    # standalone policy from this
+    standalone_policy = actor_value.get_policy_operator()
+    ```
+    </details>
+
+- exploration [wrappers](torchrl/modules/tensordict_module/exploration.py) and [modules](torchrl/modules/models/exploration.py) to easily swap between exploration and exploitation<sup>(1)</sup>:
+    <details>
+      <summary>Code</summary>
+    
+    ```python
+    policy_explore = EGreedyWrapper(policy)
+    with set_exploration_mode("random"):
+        tensordict = policy_explore(tensordict)  # will use eps-greedy
+    with set_exploration_mode("mode"):
+        tensordict = policy_explore(tensordict)  # will not use eps-greedy
+    ```
+    </details>
+
 - various [recipes](torchrl/trainers/helpers/models.py) to build models that correspond to the environment being deployed;
-- a generic [trainer class](torchrl/trainers/trainers.py).
+- a generic [trainer class](torchrl/trainers/trainers.py)<sup>(1)</sup>.
```

Comment:
bring this near the top with the custom training loop, either build your own loop or use the prebuilt one